<!DOCTYPE HTML>
<html>
   <head>
      <title>Lab of Statistical Genetics and Genomics</title>
      <meta name="description" content="Upenn Lab of Statistical Genetics and Genomics" />
      <meta name="keywords" content="University of Pennsylvania, Upenn, Lab, Statistics, Genetics, Genomics, Research, Papers, Lee, Li, HONGZHE LI" />
      <meta http-equiv="content-type" content="text/html; charset=windows-1252" />
      <link rel="stylesheet" type="text/css" href="style/style.css" title="style" />
      <link rel="stylesheet" type="text/css" href="style/custom.css" title="style" />
   </head>
   <body>
      <div id="main">
         <div id="header">
            <div id="logo">
               <div class = "logo-image-holder">
                  <image class = "logo-image" src = "images/upenn.png">
               </div>
               <div id="logo_text">
                  <!-- class="logo_colour", allows you to change the colour of the text -->
                  <h1> <a href="index.html">Lab of Statistical Genetics and Genomics</a></h1>
                  <h2>Hongzhe Li, Ph.D.</h2>
               </div>
            </div>
            <div id="menubar">
               <ul id="menu">
                  <li><a href="index.html">Home</a></li>
                  <li><a href="labpi.html">Lab Pi</a></li>
                  <li><a href="research.html">Research</a></li>
                  <li><a href="awards.html">Awards/Honors</a></li>
                  <li><a href="publications.html">Publications</a></li>
                  <li class="selected"><a href="software.html">Software</a></li>
               </ul>
            </div>
         </div>
         <div id="site_content">
           <div class="sidebar">
             <image class = "sidebar-picture" src='images/hongzhe.jpg'>
             <h1 style = "text-align: center;padding-top:0px;margin-bottom:0px">Hongzhe Li, Ph.D.</h1>
             <div class = "sidebar-titles" >
               <span >Professor of Biostatistics and Statistics </span><br>
               <span class = "title-example" >Chair, Graduate Program in Biostatistics </span><br>
               <span class = "title-example" >Director, Center for Statistics in Big Data </span><br>
               <span class = "title-example" >Fellow, ASA; Fellow, IMS; Fellow, AAAS</span><br>
             </div>
             <hr style = "margin-top:10px;">
             <div style="margin-top:10px;"class = "sidebar-titles" >
               <span style="color:#43423F;font-size:12px;"><u>Address</u></span><br>
               <span >215 Blockley Hall</span><br>
               <span>423 Guardian Drive</span><br>
               <span>University of Pennsylvania </span><br>
               <span>Philadelphia, PA 19104</span><br>
             </div>
             <hr style = "margin-top:10px;">
             <div style="margin-top:10px;"class = "sidebar-titles" >
               <span style="color:#43423F;font-size:12px;"><u>Contact</u></span><br>
               <span >Tel: (215) 573-5038</span><br>
               <span>Web: <a href="statgene.upenn.edu">statgene.upenn.edu</a></span><br><br>
               <span>Assistant: Meghan Lecates</span><br>
               <span>Tel: (215) 573-7327</span><br>
             </div>
             <hr style = "margin-top:10px;margin-bottom:10px;">
             <div style="margin-top:10px;"class = "sidebar-titles" >
               <span style="color:#43423F;font-size:12px;text-align:center;"><u>About</u></span><br>
             </div>

             <div style = "font-size:11px;text-indent: 20px;">
               <p style="line-height: 1.3em;">I did my undergradute study in mathematics at Peking University in Beijing and received a Ph.D. degree in Statistics in 1995 from the University of Washington in Seattle under the supervision of... <a href="labpi.html">Read more</a></p>
             </div>
             <p></p>

           </div>
            <div id="content">
               <h1>Software Programs from Hongzhe Li's Group</h1>
               <p>Supported by NIH grants ES009911, CA127334 and GM097505, we have developed statistical and computational methods for analysis of genetic, genomics and microbiome data. We list here the programs that implement some of our latest statistical methods, most are written in R and are available as Cran-R packages. Some datasets can also be found here. Note that these program are constantly beining updated. For questions and comments, please email Dr. Hongzhe Li, (hongzhe@upenn.edu).</p>
               <h2 style = "margin-bottom:0px;">Downloads</h2>
               <small>GUniFrac, MCLIME, HighDCox, DirMulti, HeteroReg, qMSAT, LRS, MATCHCLIP, CNVtest.</small>
               <br><br>
               <h4><a href="downloads/GLMLasso-HLi_code_submit.zip" download>Download GLM-constraints</a></h4>
               <span class = "program-description">This program .zip file includes Matlab codes and real data sets used in paper "Generalized Linear Models with Linear Constraints for Microbiome Compositional Data".</span><br>
               <span class = "program-authors">- Jiaru Lu, Pixu Shi and Li H (2018 Biometrics, revision).<span><br>
               <span class = "">- Source code language:<span><span> Mathlab</span><br>
               <br>

               <h4><a href="downloads/Matlab Codes and RealData.zip" download>Download FDE</a></h4>
               <span class = "program-description">This program .zip file includes Matlab codes and real data sets used in paper "Generalized Linear Models with Linear Constraints for Microbiome Compositional Data".</span><br>
               <span class = "program-authors">- Guo Z, Wang W, Cai, TT and Li H (2017 JASA).<span><br>
               <span class = "">- Source code language:<span><span> Mathlab</span><br>
               <br>
               <h4><a href="downloads/CodeAndData.zip" download>Download Modified Rank Tests</a></h4>
               <span class = "program-description">This program .zip file includes R codes and real data sets used in paper on modified rank tests for data with excessive zeros.</span><br>
               <span class = "program-authors">- Wang W, Chen EZ and Li H (2017 submitted).<span><br>
               <span class = "">- Source code language:<span><span> R</span><br>
               <br>

               <h4><a href="http://cran.r-project.org/web/packages/GUniFrac/index.html" target="_blank">GUniFrac</a></h4>
               <span class = "program-description">This program is available as a cran-R package <a target="_blank" href = "https://cran.r-project.org/web/packages/GUniFrac/index.html">(link here)</a> that implements a generalized UniFrac distance for analysis of microbiome data.</span><br>
               <span class = "program-authors">- Chen J and Li H (2012).<span><br>
               <span class = "">- Source code language:<span><span> R</span><br>

               <br>
               <h4><a href="downloads/MCLIME_1.1.tar copy.gz" download>Download MCLIME</a></h4>
               <span class = "program-description">This program is available as a cran-R package <a target="_blank" href="https://cran.r-project.org/src/contrib/Archive/MCLIME/">(link here)</a> that implements simultaneous estimation of the regression coefficients and precision matrix.</span><br>
               <span class = "program-authors">- Xie J, Cai T and Li H (2012).<span><br>
               <span class = "">- Source code language:<span><span> R/SPlus</span><br>

               <br>
               <h4 class = "black-header-color"><u>HighDCox</u></h4>
               <span class = "program-description">The following is a list of R programs for linking the microarray gene expression data to censroed survival data. Dataexample.R includes examples of using these functions using simulated data sets. Further details of the functions can be found in this documentmetaion (pdf file) of the programs.</span><br><br>
               <div class = "program-description">
                  Programs:
                  <ul>
                  <ol>
                     <li><a href = "">Download  PCRCox</a>
                        <br>
                        Partial Cox regression method
                        <br>
                        <span class = "program-authors">- Li H , Gui J(2004): Partial Cox Regression Analysis for High-Dimensional Microarray Gene Expression Data. Bioinformatics , 20(suppl 1), i208-i215. (<a href = "downloads/pcrcox.pdf" target="_blank">pdf file</a>)</span><br>
                        <span class = "program-authors">- Source code language: R/Splus.</span>
                     </li>
                     <li><a href = "">Download TGDCox</a>
                        <br>
                        Threshold gradient descent methods for fitting the Cox model
                        <br>
                        <span class = "program-authors">- Gui J, Li H (2005): Penalized Cox Regression Analysis in the High-Dimensional and Low-sample Size Settings, with Applications to Microarray Gene Expression Data. Bioinformatics , 21: 3001-3008.</span><br>
                        <span class = "program-authors">- Source code language: R/Splus.</span>
                     </li>
                  </ol>
               </div>



               <h4><a href="">Download DirMulti</a></h4>
               <span class = "program-description">Variable selection for Dirichlet-Multinomial regression for identifying covariates that are associated with microbiomes.</span><br>
               <span class = "program-authors">- Chen J, Li H (2012): Annals of Applied Statistics (revision).<span><br>
               <span class = "">- Source codes: R/Splus<br>
               <br>

               <h4><a href="">Download LRS</a></h4>
               <span class = "program-description">Likelihood ratio selector for identifying the copy number variants (CNVs) for the SNP Chip LogR ratio data.</span><br>
               <span class = "program-authors">- Jeng J, Cai TT and Li H (2010): Optimal sparse segment identification with application in copy number variation analysis. Journal of American Statistical Association, 105 (491): 1156-1166.<span><br>
               <span class = "">- Source codes: R/Splus<br>
               <br>

               <h4><a href="">Download HeteroReg</a></h4>
               <span class = "program-description">Variable selction for high dimesional Heteroscedastic Regression.</span><br>
               <span class = "program-authors">- Daye J, Chen J and Li H (2012): High-dimensional heteroscedastic regression with an application to eQTL data analysis Biometrics, 68: 316-326.<span><br>
               <span class = "">- Source codes: Fortran and R/Splus</span><br>

               <br>
              <h4><a href="https://web.njit.edu/~zhiwei/qMSAT/" target="_blank">qMSAT</a></h4>
               <span class = "program-description">A Quality-based Multivariate Score Association Test.</span><br>
               <span class = "program-authors">- Z. John Daye, Hongzhe Li, and Zhi Wei (2012): A Powerful Test for Multiple Rare Variants Association Studies that Incorporate Sequencing Qualities. Nucleic Acids Research, in press<span><br>

                <br>
               <h4><a href="https://github.com/yhwu/matchclips/" target="_blank">MATCHCLIP</a></h4>
               <span class = "program-description">Program for identifying the breakpoints and CNVs based on the next generation sequence data using CIGAR strings.</span><br>
               <span class = "program-authors">- Wu Y, Tian L, Pirastu M, Stambolian D, Li H(2013): MATCHCLIP: Locate precise break points for copy number variation using CIGAR strings.<span><br>

                 <br>
               <h4><a href="">Download CNVtest</a></h4>
               <span class = "program-description">R Program for testing CNV associtiob based on SNP array log-ratio data for both case-control and quantative traits.</span><br>
               <span class = "program-authors">- Jeng J, Wu Q and Li H (2015): A Statistical Method for Identifying Trait-Associated Copy Number Variants.<span><br>
                 <span class = "">- Source codes: R/Splus<br>
               <br>


            </div>
         </div>
         <div id="footer">
            Lab of Statistical Genetics and Genomics / Hongzhe Li, Ph.D. / (215) 573-5038
         </div>
      </div>
   </body>
</html>
